marlowe-runtime/test/Language/Marlowe/Runtime/Transaction/BuildConstraintsSpec.hs

```diff
@@ -223,3 +236,88 @@ withdrawSpec = Hspec.describe "buildWithdrawConstraints" do
           }
 
     pure $ actual `shouldBe` expected
+
+buildApplyInputsConstraintsSpec :: Spec
+buildApplyInputsConstraintsSpec =
+  Hspec.describe "buildApplyInputsConstraints" do
+    let
+      -- System start and era history are reusable across tests.
+      systemStart = SystemStart $ posixSecondsToUTCTime 0  -- Without loss of generality.
+      eraParams = EraParams
+        { eraEpochSize = 1
+        , eraSlotLength = mkSlotLength 1
+        , eraSafeZone = UnsafeIndefiniteSafeZone
+        }
+      oneSecondBound i = Bound
+        { boundTime = RelativeTime $ fromInteger i
+        , boundSlot = fromInteger i
+        , boundEpoch = fromInteger i
+        }
+      oneSecondEraSummary i = EraSummary
+        { eraStart = oneSecondBound i
+        , eraEnd = EraEnd $ oneSecondBound $ i + 1
+        , eraParams
+        }
+      unboundedEraSummary i = EraSummary
+        { eraStart = oneSecondBound i
+        , eraEnd = EraUnbounded
+        , eraParams
+        }
+      eraHistory =
+        EraHistory CardanoMode
+          . mkInterpreter
+          . summaryWithExactly
+          $ Exactly
+          $  K (oneSecondEraSummary 0) -- Byron lasted 1 second
+          :* K (oneSecondEraSummary 1) -- Shelley lasted 1 second
+          :* K (oneSecondEraSummary 2) -- Allegra lasted 1 second
+          :* K (oneSecondEraSummary 3) -- Mary lasted 1 second
+          :* K (oneSecondEraSummary 4) -- Alonzo lasted 1 second
+          :* K (unboundedEraSummary 5) -- Babbage never ends
+          :* Nil
+    Hspec.QuickCheck.prop "valid slot interval for timed-out contract" do
+      address <- arbitrary
+      assets <- arbitrary
+      utxo <- arbitrary
```

Comment:
The other generated values look fine because they just generate integers which don't really benefit from shrinking, but `assets` and `utxo` could definitely benefit from shrinking here. Can you rewrite this part of the test so that these will be shrunk if the test fails?

```suggestion
    Hspec.QuickCheck.prop "valid slot interval for timed-out contract" \assets utxo -> do
      address <- arbitrary
```